### DISCUSSION TITLE
setState to Stream Deck+

### DISCUSSION THREAD

**Discussion by sn007b9368** [ybizeul/StreamDeckWS#30] [General]
I've been playing around with WSproxy on my Stream Deck+ and I've been trying to change the layout of the touchscreen according to the documentation. If I change the msg.payload with the example expression from the Elgato sdk docs such that:

{
    "event": "setFeedbackLayout",
    "context": context,
    "payload": {
        "layout": "$A0"
    }
}

When this is passed to the sd-output the event doesn't seem to change to setFeedbackLayout. Using a debug node on the output of the sd-output node shows that the event stays as setState. The context and the layout payload get passed to the Stream Deck+. Is it even possible to change the layout of the touchscreen using the setFeedbackLayout event, or am I just limited to setState, setImage, and setTitle?

I tested to see if I could send an image to the touchscreen using the setImage event but nothing seems to happen. I've been able to send images to standard buttons using the setImage event but not the touchscreen.

Your help would be much appreciated. Thanks

**Comment by n61ab** [comment]
sn007

You do not say if you using the context string for the value "context" in the key-value pair "context": context"?

As part of the object returned when the SD button is first seen "willAppear" event, it sends a unique value for "context."  You need to use this value in the JSON for "setFeedbackLayout", "context" : context: key-value pair.

This is explained in the read-me for the plugin.

Alan

**Comment by sn007b9368** [comment]
n61ab
thanks for your reply

yes, i'm using the context string for the key-value pair, but for some reason it doesn't seem to be getting through.

I'm just learning node red for the first time today and it's a fairly steep learning curve.

**Comment by sn007b9368** [comment]
I can either get it to work where the `context`  is sent through to the stream deck but the state remains as setState, or with a bit of contortion, I can get setState to change to setFeedbackLayout but the `context` does not get through and the debug shows it as _undefined_.

**Comment by n61ab** [comment]
Here is my Node-Red flow built to explore the SD+

Change the file type to ."json"

See my bug report.  I have both Dials and Button.  At this time, you canont use both a Dial and Button on the same SD+

Alan


[Dials and Buttons.txt](https://github.com/ybizeul/StreamDeckWS/files/10911328/Dials.and.Buttons.txt)

**Comment by sn007b9368** [comment]
Thanks very much, i'll go away and analyze how yours works.

I just managed to get the `state` to change to setFeedbackLayout and the 'context' to the unique id and the layout payload is being sent, but the physical touchscreen layout on the Stream Deck+ isn't changing, so I'm still doing something wrong. I'll keep plugging away...
